src/geode/mesh/builder/solid_mesh_builder.cpp

```diff
@@ -280,6 +280,76 @@ namespace
             builder.create_polyhedron( vertices, facets );
         }
     }
+
+    template < geode::index_t dimension >
+    void update_edge_and_facet( const geode::SolidMesh< dimension >& solid,
+        geode::SolidMeshBuilder< dimension >& builder,
+        const geode::PolyhedronVertex& polyhedron_vertex,
+        geode::index_t new_vertex_id )
+    {
+        for( const auto f : geode::LRange{ solid.nb_polyhedron_facets(
+                 polyhedron_vertex.polyhedron_id ) } )
+        {
+            const geode::PolyhedronFacet id{ polyhedron_vertex.polyhedron_id,
+                f };
+            const auto nb_facet_vertices =
+                solid.nb_polyhedron_facet_vertices( id );
+            absl::FixedArray< geode::PolyhedronVertex > facet_vertices(
+                nb_facet_vertices );
+            for( const auto v : geode::LRange{ nb_facet_vertices } )
+            {
+                facet_vertices[v] =
+                    solid.polyhedron_facet_vertex_id( { id, v } );
+            }
+            const auto position_it =
+                absl::c_find( facet_vertices, polyhedron_vertex );
+            if( position_it != facet_vertices.end() )
+            {
+                geode::PolyhedronFacetVertices facet_vertices_id;
+                facet_vertices_id.reserve( nb_facet_vertices );
+                for( const auto& v : facet_vertices )
+                {
+                    facet_vertices_id.emplace_back(
+                        solid.polyhedron_vertex( v ) );
+                }
+
+                const auto position = static_cast< geode::index_t >(
+                    std::distance( facet_vertices.begin(), position_it ) );
+
+                if( solid.are_facets_enabled() )
```

Comment:
Maybe skip at thé begining of the function if both edges and facets are not enabled